controllers/articles.controllers.js

```diff
@@ -18,30 +18,36 @@ exports.getEndpoints = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  Promise.all([fetchArticles(), fetchComments()]).then((promises) => {
-    const articles = promises[0];
-    const comments = promises[1];
-    const newArticles = articles.map((article) => {
-      let commentCount = 0;
-      comments.forEach((comment) => {
-        if (comment.article_id === article.article_id) {
-          commentCount += 1;
-        }
+  const { topic } = req.query;
+  Promise.all([fetchArticles(topic), fetchComments()])
+    .then((promises) => {
+      const articles = promises[0];
+      const comments = promises[1];
+      const newArticles = articles.map((article) => {
+        let commentCount = 0;
+        comments.forEach((comment) => {
+          if (comment.article_id === article.article_id) {
```

Comment:
This is fine but you can use an aggregate function within your SQL query that would do all of this for you. I would suggest that you go back at some point in the future and refactor(not now)